plugins/main/public/components/overview/vulnerabilities/dashboards/inventory/inventory.tsx

```diff
@@ -127,56 +129,69 @@ const InventoryVulsComponent = () => {
         grow
       >
         <>
-          {isLoading ? 
-            <LoadingSpinner /> : 
-            <SearchBar 
-              appName='inventory-vuls' 
-              {...searchBarProps} 
+          {isLoading ?
+            <LoadingSpinner /> :
+            <SearchBar
+              appName='inventory-vuls'
+              {...searchBarProps}
               showDatePicker={false}
               showQueryInput={true}
               showQueryBar={true}
-              />}
-          {isSearching ? 
+            />}
+          {isSearching ?
             <LoadingSpinner /> : null}
-          {!isLoading && !isSearching && results?.hits?.total === 0 ? 
+          {!isLoading && !isSearching && results?.hits?.total === 0 ?
             <DiscoverNoResults timeFieldName={timeField} queryLanguage={''} /> : null}
-          {!isLoading && !isSearching && results?.hits?.total > 0 ?
-            <EuiDataGrid 
-              {...dataGridProps} 
+          {!isLoading && !isSearching && results?.hits?.total > 0 ? (
+            <EuiDataGrid
+              {...dataGridProps}
               toolbarVisibility={{
                 additionalControls: (
-                  <EuiButtonEmpty
-                    disabled={results?.hits?.total === 0}
-                    size="xs"
-                    iconType="exportAction"
-                    color="primary"
-                    className="euiDataGrid__controlBtn"
-                    onClick={onClickExportResults}>
-                    Export Formated
-                  </EuiButtonEmpty>
-                ),
+                  <>
+                    <HitsCounter
+                      hits={results?.hits?.total}
+                      showResetButton={false}
+                      onResetQuery={() => { }}
+                      tooltip={results?.hits?.total && results?.hits?.total > MAX_ENTRIES_PER_QUERY ? {
+                        ariaLabel: 'Warning',
+                        content: `The query results has exceeded the limit of 10,000 hits. To provide a better experience the table only shows the first ${formatNumWithCommas(MAX_ENTRIES_PER_QUERY)} hits.`,
+                        iconType: 'alert',
+                        position: 'top'
+                      } : undefined}
+                    />
+                    <EuiButtonEmpty
+                      disabled={results?.hits?.total === 0}
+                      size="xs"
```

Comment:
I suggest to use the prop "isLoading" with a state:
```
isLoading={isExporting}
```
And set a status in the function `onClickExportResults`:

```
    try {
      setIsExporting(true)
      await exportSearchToCSV(params);
    } catch (error) {
      const searchError = ErrorFactory.create(HttpError, { error, message: 'Error downloading csv report' })
      ErrorHandler.handleError(searchError);
    } finally {
      setIsExporting(false)
    }
```